cpp/src/io/json/json_column.cu

```diff
@@ -530,13 +530,15 @@ void make_device_json_column(device_span<SymbolT const> input,
   auto max_row_offsets = cudf::detail::make_std_vector_async(d_max_row_offsets, stream);
   std::vector<std::string> column_names = copy_strings_to_host(
     input, d_column_tree.node_range_begin, d_column_tree.node_range_end, stream);
+  stream.synchronize();
   // array of arrays column names
   if (is_array_of_arrays) {
     TreeDepthT const row_array_children_level = is_enabled_lines ? 1 : 2;
     auto values_column_indices =
       get_values_column_indices(row_array_children_level, tree, col_ids, num_columns, stream);
     auto h_values_column_indices =
       cudf::detail::make_std_vector_async(values_column_indices, stream);
+    stream.synchronize();
```

Comment:
Then `make_std_vector_sync` above this.

Reply:
I was bit sceptical about this because, if any optimization in future is done inside `make_std_vector_sync` to return empty vector without sync, this will cause async issues. So, to be safe, it added `stream.synchronize();` outside. Anyway, the sync code does the same.